jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/SecureRequestCustomizer.java

```diff
@@ -230,6 +232,31 @@ protected void checkSni(Request request, SSLSession session)
         }
     }
 
+    protected String retrieveSni(Request request, SSLSession session)
+    {
+        // Quick retrieval of the SNI from a SSLSession attribute put by SniX509ExtendedKeyManager.
+        String sniHost = (String)session.getValue(SslContextFactory.Server.SNI_HOST);
+        if (sniHost != null)
+            return null;
+
+        // Some security providers (for example, Conscrypt) do not support
+        // SSLSession attributes, so perform a more expensive SNI retrieval.
+        if (session instanceof ExtendedSSLSession extended)
+        {
+            sniHost = extended.getRequestedServerNames().stream()
+                .filter(SNIHostName.class::isInstance)
+                .map(SNIHostName.class::cast)
+                .map(SNIHostName::getAsciiName)
+                .findFirst()
+                .orElse(null);
+            if (sniHost != null)
+                return sniHost;
+        }
```

Comment:
I don't think streams add any clarity here.  The code below is clearer and more efficient:

```suggestion
        if (session instanceof ExtendedSSLSession extended)
        {
            for (SNIServerName serverName : extended.getRequestedServerNames())
            {
                if (serverName instanceof SNIHostName hostName)
                    return hostName.getAsciiName();
            }
        }
```